### DISCUSSION TITLE
Everyday Usage

### DISCUSSION THREAD

**Discussion by purgat0ri** [Homebrew/discussions#5501] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c98a33a376e5b35ebc801d0f7d6ba7b9eb118ac
Last commit: 10 days ago
Core tap JSON: 16 Jul 01:18 UTC
Core cask tap JSON: 16 Jul 01:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: XXXX
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.45.2 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1.1-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  cakebrew

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.4.


Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-fonts
Untap them with `brew untap`.

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.
```

### Description of issue

Error encountered when attempting to run telega-server:

TDLib version=1.8.25 < 1.8.29 (min required)
Please upgrade TDLib and recompile `telega-server'

Is there any way to install TDLib 1.8.29?

**Comment by SMillerDev** [comment]
> Error encountered when attempting to run telega-server:
```console
TDLib version=1.8.25 < 1.8.29 (min required)
Please upgrade TDLib and recompile `telega-server'
```

Could you provide some instructions how to reproduce this?

**Reply by purgat0ri** [reply]
Sure. 

1. Install docker
2. Install TDLib
3. Install Telega server in Docker:

`$ docker pull zevlg/telega-server:latest`

4. Install Telega from MELPA

5. Setup Telega in Emacs:

```
;; Telega - Unofficial Emacs Telegram client
(use-package! telega
  :defer t
  :bind (:map evil-normal-state-map
         ("SPC t v" . telega)
         :map telega-prefix-map
         ("g" . telega-filter-by-folder))
  :config
  (require 'telega-alert)
  (setq telega-server-libs-prefix "/opt/homebrew")
  (setq telega-chat-fill-column 190)
  (setq telega-use-docker t)
  (setq telega-accounts (list
                         (list "demonnsd" 'telega-database-dir telega-database-dir)
                         (list "artawower" 'telega-database-dir (expand-file-name "artawower" telega-database-dir))
                         (list "ksenofobious" 'telega-database-dir (expand-file-name "ksenofobious" telega-database-dir))))
  (telega-alert-mode 1))

```


6. Run M-x 'telega' in Emacs (Emacs-plus, 29.4), which calls telega-server.
7. The above error is returned during telega-server startup and the process terminates.

**Reply by SMillerDev** [reply]
Okay, so that has nothing to do with brew then since the server is in docker.

**Reply by purgat0ri** [reply]
> Okay, so that has nothing to do with brew then since the server is in docker.

Right. But the server depends on TDLib, and the version in Brew seems to be outdated? (Apologies, I don't have a great understanding of Brew, so I'm not sure if the latest version *is* available, and I just don't know how to install it).

**Reply by SMillerDev** [reply, marked as answer]
But it being in docker means it should have a tdlib install of it's own in docker. The version in brew might be outdated, but there is a clearly defined border here. Tools inside docker can't talk to tools outside docker.

**Reply by purgat0ri** [reply]
Ahhh, I see. Thanks for that. I was under the mistaken impression that telega-server in Docker depended on the TDLib installed by Brew (I also don't understand Docker very well). Once you set me right, though, all it took was reinstalling the Docker image to resolve the problem. Cheers again.
